gc/gc-iceberg/src/main/java/org/projectnessie/gc/iceberg/IdentifyExpiredSnapshotsProcedure.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.iceberg;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.iceberg.spark.procedures.BaseGcProcedure;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.catalyst.util.MapData;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.iceberg.catalog.ProcedureParameter;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.projectnessie.gc.base.GCImpl;
+import org.projectnessie.gc.base.GCUtil;
+import org.projectnessie.gc.base.ImmutableGCParams;
+
+/**
+ * Nessie GC procedure to identify the expired snapshots per content id per reference via the {@link
+ * org.projectnessie.gc.base.GCImpl#identifyExpiredContents(SparkSession)} functionality.
+ *
+ * <p>Writes the identified output to the Iceberg table via {@link
+ * org.projectnessie.gc.base.IdentifiedResultsRepo}.
+ *
+ * <p>Accepts the {@link org.projectnessie.gc.base.GCParams} members as the arguments and returns
+ * the run id of the completed GC task which can be used to query the results stored in the Iceberg
+ * table via {@link org.projectnessie.gc.base.IdentifiedResultsRepo}.
+ */
+public class IdentifyExpiredSnapshotsProcedure extends BaseGcProcedure {
```

Comment:
Detail documentation will be added after both `identify()` and `expire()` call procedure is added. Tracker: https://github.com/projectnessie/nessie/issues/2020